### PR TITLE
fix(card-in-card): prevent new tab from opening twice on Firefox

### PR DESCRIPTION
**Pull request by proeung**
### Related Ticket(s)

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/7941

### Description

- This PR modifies the `onclick` function to query select the `dds-card-in-card-image` instead of the whole card.

### Changelog

**New**

- Adjust the WC sandbox example to reflect the up to date markup.

**Test**

- Click on the WC sandbox link and navigate to the `web-components/examples/codesandbox/components/card-in-card`.
- Open this sandbox preview in FF
- Click on the CTA card located on the bottom right-hand corner.
- Ensure that the new tab is opening just once.

### CODE REVIEW

File: packages/web-components/src/components/card-in-card/card-in-card.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ class DDSCardInCard extends StableSelectorMixin(DDSCardCTA) {
     if (linkNode) {
       linkNode.classList.add(`${prefix}--card-in-card`);
     }
-    this.onclick = () =>
+    this.querySelector('dds-card-in-card-image').onclick = () =>
```
Comment (ariellalgilmore) [inline packages/web-components/src/components/card-in-card/card-in-card.ts:51]:
```suggestion
    this.querySelector('dds-card-in-card-image')?.onclick = () =>
```

Comment (proeung) [reply packages/web-components/src/components/card-in-card/card-in-card.ts]: @ariellalgilmore Not sure if this ternary operator suggestion works here, but I went ahead and added `if` conditional check. let me know what you think.

Comment (ariellalgilmore) [reply packages/web-components/src/components/card-in-card/card-in-card.ts]: i think we can stick with the ternary but looks like ci-checks don't like the `onclick`, so let's try this

```
this.querySelector(`${ddsPrefix}-card-in-card-image`)?.addEventListener('onclick', () => {
      this.querySelector(`${ddsPrefix}-card-cta-footer`)
        ?.shadowRoot?.querySelector(`a`)
        ?.click();
    });
```

**Comment by proeung** [comment]
@ariellalgilmore Added your suggestion and this PR is ready for a re-review!

**Review by ariellalgilmore — APPROVED** [review]
LGTM!

**Pull request merged**